x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -514,6 +514,10 @@ export class Plugin implements ISecuritySolutionPlugin {
 
     // Assistant Tool and Feature Registration
     plugins.elasticAssistant.registerTools(APP_UI_ID, getAssistantTools());
+    plugins.elasticAssistant.registerFeatures(APP_UI_ID, {
+      assistantModelEvaluation: config.experimentalFeatures.assistantModelEvaluation,
```

Comment:
The Security Solution happens to be using the Security Solution-Specific `experimentalFeatures` abstraction here, when registering the `assistantModelEvaluation` feature, because the `experimentalFeatures` abstraction has several benefits.

It's worth noting that a plugin is free to use _any_ implementation of reading feature flags when registering a feature with the assistant. (There is no requirement to use something like the `experimentalFeatures` abstraction in other plugins when registering a feature with the assistant.)

Reply:
Bingo, well stated! And right now there is a minor feature-gap with the assistant plugin itself as it has no 'kibana.yml' configuration keys registered for setting these values, and so capabilities must be registered by the consuming plugin. 

That said (and as we discussed offline), would love to have a streamlined way of managing configuration/capabilities. I have a few kibana platform docs to wade through to see what they're up to first before proposing anything.